mathesar/api/serializers/shared_serializers.py

```diff
@@ -106,9 +108,106 @@ class NumberDisplayOptionSerializer(OverrideRootPartialMixin, serializers.Serial
     locale = serializers.CharField(required=False)
 
 
+class AbstractDateTimeFormatValidator(ABC):
+    requires_context = True
+
+    def __init__(self):
+        pass
+
+    def __call__(self, value, serializer_field):
+        self.date_format_validator(value)
+
+    def date_format_validator(self, value):
+        try:
+            timestamp_with_tz_obj = arrow.get('2013-09-30T15:34:00.000-07:00')
+            parsed_datetime_str = timestamp_with_tz_obj.format(value)
+            datetime_object = arrow.get(parsed_datetime_str, value)
+        except ValueError:
+            raise serializers.ValidationError(f"{value} is not a valid format used for parsing a datetime.")
+        else:
+            self.validate(datetime_object, value)
+
+    @abstractmethod
+    def validate(self, datetime_obj, display_format):
```

Comment:
> The validation is spread over two functions called in different ways, which is a bit confusing. Couldn't one call the other?

@mathemancer Do you mean this validation function?

Reply:
Whoops.  I completely missed the `else`.  My fault.  disregard that.